core/io/resource_format_binary.cpp

```diff
@@ -426,14 +436,29 @@ Error ResourceLoaderBinary::parse_variant(Variant &r_v) {
 						path = ProjectSettings::get_singleton()->localize_path(res_path.get_base_dir().path_join(path));
 					}
 
-					if (remaps.find(path)) {
+					if (remaps.has(path)) {
 						path = remaps[path];
 					}
 
-					Ref<Resource> res = ResourceLoader::load(path, exttype, cache_mode_for_external);
+					if (using_whitelist && !external_path_whitelist.has(path)) {
+						WARN_PRINT(vformat("Blocked path not on whitelist: %s.", path));
+						r_v = Variant();
+						error = ERR_FILE_CANT_OPEN;
+						return error;
+					}
+
+					Ref<Resource> res;
+					if (using_whitelist) {
+						res = ResourceLoader::load_whitelisted(path, external_path_whitelist, type_whitelist, exttype, cache_mode_for_external);
```

Comment:
We should not apply the whitelist to loading external resources, since we have already whitelisted this resource if we are loading it